src/log-store/src/kafka/periodic_offset_fetcher.rs

```diff
@@ -0,0 +1,122 @@
+// Copyright 2023 Greptime Team
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+use std::time::Duration;
+
+use common_telemetry::{debug, error, info};
+use tokio::time::{interval, MissedTickBehavior};
+
+use crate::error::Result;
+use crate::kafka::client_manager::ClientManagerRef;
+
+/// PeriodicOffsetFetcher is responsible for periodically updating the offset
+/// for each Kafka topic.
+pub(crate) struct PeriodicOffsetFetcher {
+    /// Interval to fetch the latest offset.
+    interval: Duration,
+    /// Client manager to send requests.
+    client_manager: ClientManagerRef,
+}
+
+impl PeriodicOffsetFetcher {
+    pub(crate) fn new(interval: Duration, client_manager: ClientManagerRef) -> Self {
+        Self {
+            interval,
+            client_manager,
+        }
+    }
+
+    /// Starts the offset fetcher as a background task
+    ///
+    /// This spawns a task that periodically queries Kafka for the latest
+    /// offset values for all registered topics and updates the shared map.
+    pub(crate) async fn run(self) {
+        info!("PeriodicOffsetFetcher started");
+        common_runtime::spawn_global(async move {
+            let mut interval = interval(self.interval);
```

Comment:
log when it actually starts.
```suggestion
        common_runtime::spawn_global(async move {
            info!("PeriodicOffsetFetcher started");
            let mut interval = interval(self.interval);
```